admin-dev/themes/new-theme/js/pages/product/combination/bulk-form-handler.ts

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ConfirmModal from '@components/modal';
+import ProductMap from '@pages/product/product-map';
+import ComponentsMap from '@components/components-map';
+import ProductEvents from '@pages/product/product-event-map';
+import CombinationsService from '@pages/product/services/combinations-service';
+import {EventEmitter} from 'events';
+import DisablingToggler from '@components/form/disabling-toggler';
+
+const CombinationMap = ProductMap.combinations;
+const CombinationEvents = ProductEvents.combinations;
+
+export default class BulkFormHandler {
+  private combinationsService: CombinationsService;
+
+  private eventEmitter: EventEmitter;
+
+  private tabContainer: HTMLDivElement;
+
+  constructor() {
+    this.combinationsService = new CombinationsService();
+    this.eventEmitter = window.prestashop.instance.eventEmitter;
+    this.tabContainer = document.querySelector(CombinationMap.externalCombinationTab) as HTMLDivElement;
+    this.init();
+  }
+
+  private init(): void {
+    this.listenSelections();
+    this.eventEmitter.on(CombinationEvents.listRendered, () => this.toggleBulkActions());
+
+    const bulkFormBtn = document.querySelector(CombinationMap.bulkCombinationFormBtn) as HTMLButtonElement;
+    bulkFormBtn.addEventListener('click', () => this.showFormModal());
+  }
+
+  private showProgressModal(): ConfirmModal {
+    //@todo: Replace with new progress modal when introduced in #26004.
+    const modal = new ConfirmModal(
+      {
+        id: CombinationMap.bulkProgressModalId,
+        confirmMessage: '<div>Updating combinations: <p class="progress-increment"></p></div>',
+      },
+      () => null,
+    );
+
+    modal.show();
+
+    return modal;
+  }
+
+  private showFormModal(): ConfirmModal {
+    const template = document.querySelector(CombinationMap.bulkCombinationFormTemplate) as HTMLScriptElement;
+    const content = template.innerHTML;
```

Comment:
Just a suggestion, but maybe this could be fetched and stored in a class field when the component is constructed